WordPress/gutenberg
# Image Block: Enable image block to be selected correctly when clicked

Follow-up on #55470
Fixes: #56017

## What?

This PR allows you to select image blocks "correctly".

This will display a blue outline when a block is selected, and inserting a block with the Enter key and deleting a block with the Delete key will work correctly.

## Why?

With #55470, the image block is now wrapped in an A tag when it's linked. As a result, even if you click on the block, it no longer appears to be selected correctly.

## How?

To find a way to solve this, I looked into blocks where the block itself is an A element, or where there is an A element inside the block. As a result, I noticed that there are many approaches that add `pointer-events:none` to the A element.

I can't explain it exactly, but I found that block selection works correctly with this CSS.

**Update:** I found that if I simply added `pointer-events: none` to the `a` element, the image could no longer be resized. This is because the `ResizableBox` component is wrapped in an `a` element. To solve this I moved the `a` element inside the `ResisableBox`. Check [this comment](https://github.com/WordPress/gutenberg/pull/56043#issuecomment-1810356718) for details.

## Testing Instructions

- Insert an Image block.
- Set a link to the Image block.
- Remove focus from the Image block.
- "Click" on the image block (Note: If you select the image block with the up and down keys on your keyboard, the block will be selected correctly, so click on the block with your mouse).
- A blue outline will appear to indicate that the block is selected.
- When you press Enter, an empty paragraph block will be generated next to the image block.
- You can delete the image block by pressing the Delete key.

## Screenshots or screencast

https://github.com/WordPress/gutenberg/assets/54422211/8a7a1d8d-2c4c-4f46-9c79-1109b05adfec
